webapi/src/main/scala/org/knora/webapi/routing/admin/ProjectsRouteADM.scala

```diff
@@ -141,13 +152,36 @@ class ProjectsRouteADM(routeData: KnoraRouteData)
   private def addProject(featureFactoryConfig: FeatureFactoryConfig): Route = path(ProjectsBasePath) {
     post {
       entity(as[CreateProjectApiRequestADM]) { apiRequest => requestContext =>
+        val maybeLongname: Option[Longname] = apiRequest.longname match {
+          case Some(value) => Some(Longname.create(value).fold(error => throw error, value => value))
+          case None        => None
+        }
+
+        val maybeLogo: Option[Logo] = apiRequest.logo match {
+          case Some(value) => Some(Logo.create(value).fold(error => throw error, value => value))
+          case None        => None
+        }
+
+        val project: ProjectCreatePayloadADM =
+          ProjectCreatePayloadADM.create(
+            id = stringFormatter
+              .validateAndEscapeOptionalProjectIri(apiRequest.id, throw BadRequestException(s"Invalid user IRI")),
+            shortname = Shortname.create(apiRequest.shortname).fold(error => throw error, value => value),
+            shortcode = Shortcode.create(apiRequest.shortcode).fold(error => throw error, value => value),
+            longname = maybeLongname,
+            description = Description.create(apiRequest.description).fold(error => throw error, value => value),
+            keywords = Keywords.create(apiRequest.keywords).fold(error => throw error, value => value),
+            logo = maybeLogo,
+            status = Status.create(apiRequest.status).fold(error => throw error, value => value),
+            selfjoin = Selfjoin.create(apiRequest.selfjoin).fold(error => throw error, value => value)
+          )
         val requestMessage: Future[ProjectCreateRequestADM] = for {
           requestingUser <- getUserADM(
             requestContext = requestContext,
             featureFactoryConfig = featureFactoryConfig
           )
         } yield ProjectCreateRequestADM(
-          createRequest = apiRequest.validateAndEscape,
+          createRequest = project,
```

Comment:
having project here is confusing, i.e., unexpected. code should never surprise.

Reply:
In typed languages it isn't that bad, because you always see what is under the variable due the type, so this can be even just `p` ;)

Reply:
impairs readability. I'm doing review exclusively in the browser, so no types.

Reply:
of course it works and can be anything. the compiler doesn't care. but code is read by humans.